repo/fsrepo/migrations/httpfetcher.go

```diff
@@ -94,11 +189,69 @@ func (f *HttpFetcher) Fetch(ctx context.Context, filePath string) ([]byte, error
 	} else {
 		rc = resp.Body
 	}
-	defer rc.Close()
 
-	return io.ReadAll(rc)
+	return rc, nil
 }
 
-func (f *HttpFetcher) Close() error {
-	return nil
+func carStreamToFileBytes(ctx context.Context, r io.ReadCloser, imPath path.ImmutablePath) ([]byte, error) {
+	defer r.Close()
```

Comment:
I think we should extract this code into either of:

1. As a Boxo example on "how to download a CAR from the gateway and verify its contents"; or
2. Make a package that does it. There's a lot of "bootstrap" code here that feels a bit annoying to write and having an exported function that does this somewhere in Boxo could be very helpful.

Reply:
Creating an example is good enough, but this is so popular use case we should have a high level boxo lib/API.

Unsure about putting this in gateway package, feels like something that deserves own.

Given we have https://www.npmjs.com/package/@helia/verified-fetch that basically is a similar abstraction, maybe we could have `boxo/fetch` and `VerifiedFetch` that works in similar way? 

We don't have to do it now, ok to do example now +  fill issue for the API.